generators/csharp/model/src/object/ObjectGenerator.ts

```diff
@@ -28,6 +29,37 @@ export class ObjectGenerator extends FileGenerator<CSharpFile, ModelCustomConfig
         this.exampleGenerator = new ExampleGenerator(context);
     }
 
+    private maybeGetLiteral({ typeReference }: { typeReference: TypeReference }): string | boolean | undefined {
+        if (typeReference.type === "container" && typeReference.container.type === "literal") {
+            const literal = typeReference.container.literal;
+            switch (literal.type) {
+                case "string":
+                    return literal.string;
+                case "boolean":
+                    return literal.boolean;
+                default:
+                    return undefined;
+            }
+        }
+        if (typeReference.type === "named") {
+            const typeDeclaration = this.context.getTypeDeclarationOrThrow(typeReference.typeId);
+            if (typeDeclaration.shape.type === "alias" && 
+                typeDeclaration.shape.resolvedType.type === "container" &&
+                typeDeclaration.shape.resolvedType.container.type === "literal") {
+                const literal = typeDeclaration.shape.resolvedType.container.literal;
+                switch (literal.type) {
+                    case "string":
+                        return literal.string;
+                    case "boolean":
+                        return literal.boolean;
+                    default:
+                        return undefined;
+                }
+            }
+        }
+        return undefined;
+    }
```

Comment:
We can recursively call the method to prevent duplicative logic.
```suggestion
    private maybeGetLiteral({ typeReference }: { typeReference: TypeReference }): string | boolean | undefined {
        if (typeReference.type === "container" && typeReference.container.type === "literal") {
            const literal = typeReference.container.literal;
            switch (literal.type) {
                case "string":
                    return literal.string;
                case "boolean":
                    return literal.boolean;
                default:
                    return undefined;
            }
        }
        if (typeReference.type === "named") {
            const typeDeclaration = this.context.getTypeDeclarationOrThrow(typeReference.typeId);
            if (typeDeclaration.shape.type === "alias") {
                return this.maybeGetLiteral(typeDeclaration.shape.resolvedType);
            }
        }
        return undefined;
    }
```